### PR TITLE
build: remove unused `<depend>` in package.xml files

### PR DESCRIPTION
**Pull request by VRichardJP**
## Description

Cleaned all unused `<depend>` reported by the ad-hoc [`unused_deps.sh`](https://github.com/autowarefoundation/autoware/issues/3468) script.

In most cases, removing the `<depend>` line is enough. But sometimes downstream packages are impacted and must add the missing dependency.
Sometimes `<depend>` are just `<build_depend>` or `<test_depend>`.

## Related links

https://github.com/autowarefoundation/autoware/issues/3468

## Tests performed

Can compile with tests and run demo with the planning simulator.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by soblin — APPROVED** [review]

File: perception/bytetrack/package.xml
[the remark this answers is not in the document]
Comment (VRichardJP) [reply perception/bytetrack/package.xml:1]: Nice catch!

I just realized Eigen is sometimes included with `#include <eigen3/Eigen/XXX>`, and sometimes with `#include <Eigen/XXX>`:
```bash
$ grep -rlI "#include .Eigen/" src | wc -l
61
$ grep -rlI "#include .eigen3/" src | wc -l
50
```

I think the `<Eigen/*>` is the normal way to include. The "eigen3" uses the fact `/usr/include` is in gcc include path by default:
```bash
$ echo | gcc -Wp,-v -x c++ - -fsyntax-only
ignoring duplicate directory "/usr/include/x86_64-linux-gnu/c++/11"
ignoring nonexistent directory "/usr/local/include/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/include-fixed"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/include/c++/11
 /usr/include/x86_64-linux-gnu/c++/11
 /usr/include/c++/11/backward
 /usr/lib/gcc/x86_64-linux-gnu/11/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
```

When exporting the compile commands, you can see `kalman_filter` is compiled with path `/usr/bin/eigen3`, but not `/usr/include` (implicit from gcc):
```json
[
{
  "directory": "/home/sig/autoware/build/kalman_filter",
  "command": "/usr/bin/c++ -DBOOST_ALLOW_DEPRECATED_HEADERS -DROS_DISTRO_HUMBLE -Dkalman_filter_EXPORTS -I/home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/include -isystem /usr/include/eigen3 -fdiagnostics-color -O2 -g -DNDEBUG -fPIC -Wall -Wextra -Wpedantic -Werror -std=c++17 -o CMakeFiles/kalman_filter.dir/src/kalman_filter.cpp.o -c /home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/src/kalman_filter.cpp",
  "file": "/home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/src/kalman_filter.cpp"
},
{
  "directory": "/home/sig/autoware/build/kalman_filter",
  "command": "/usr/bin/c++ -DBOOST_ALLOW_DEPRECATED_HEADERS -DROS_DISTRO_HUMBLE -Dkalman_filter_EXPORTS -I/home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/include -isystem /usr/include/eigen3 -fdiagnostics-color -O2 -g -DNDEBUG -fPIC -Wall -Wextra -Wpedantic -Werror -std=c++17 -o CMakeFiles/kalman_filter.dir/src/time_delay_kalman_filter.cpp.o -c /home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/src/time_delay_kalman_filter.cpp",
  "file": "/home/sig/autoware/src/universe/autoware.universe/common/kalman_filter/src/time_delay_kalman_filter.cpp"
}
]
```

In other words, if someone writes "#include <eigen3/Eigen/XXX>", then `<depend>eigen</depend>` or `find_package(Eigen)` are not used at all. I don't think this is what we want (but that's not for that PR)

In any case, I will restore the `<depend>eigen</depend>` for kalman_filter and other packages.

**Review by yukkysaito — APPROVED** [review]

**Review by satoshi-ota — APPROVED** [review]

File: perception/bytetrack/package.xml
[the remark this answers is not in the document]
Comment (VRichardJP) [reply perception/bytetrack/package.xml:1]: Regarding the 2 packages you mentioned:
- [bytetrack_visualizer_node](https://github.com/autowarefoundation/autoware.universe/blob/main/perception/bytetrack/include/bytetrack/bytetrack_visualizer_node.hpp) uses `sensor_msgs::msg::Image` but does not include any `sensor_msgs` header anywhere. The code compiles because the header must be included with another message, most likely from one `tier4_perception_msgs`. I can keep the `<depend>sensor_msgs</depend>` but keep in mind it has actually no effect (brought already by another dependency) and is not used directly.
- as for `kalman_filter`, this PR does not remove the dependency on eigen. Actually, this time, eigen is marked as [`<build_depend>`](https://github.com/autowarefoundation/autoware.universe/blob/ff7c80816f09bd88a75a04c4522bcfa1c693c367/common/kalman_filter/package.xml#L17), which is questionable. Although eigen is an header only library , downstream packages won't get dependency on eigen if it is not marked as `<build_export_depend>`. I can't imagine a reasonable use case where it could cause a problem. (But that would be another topic again ^^)

**Review by kyoichi-sugahara — APPROVED** [review]

File: perception/bytetrack/package.xml
[the remark this answers is not in the document]
Comment (manato) [reply perception/bytetrack/package.xml:1]: @VRichardJP 
Thank you for your detailed comments. 
- Regarding `sensor_msgs::msg::Image`, I guess the related headers are included from `image_transport` as well as `tier4_perception_msgs` as you mentioned. I think those dependencies are reasonable and the direct dependence on the `sensor_msgs` can be removed from this package.xml.
- I appreciate your handling regarding the eigen. I agree that the proper mark for eigen is another topic, so it looks good to me about this Pull Request.

Since my all concerns are resolved, I'll approve this PullReq regarding this file. Again, thank you for your contribution!

**Review by manato — APPROVED** [review]
LGTM regarding `perception/bytetrack/package.xml`. Thx!

**Review by yukke42 — APPROVED** [review]

**Review by YamatoAndo — APPROVED** [review]

File: common/component_interface_specs/package.xml [hunk 1]
```diff
@@ -14,9 +14,10 @@
   <buildtool_depend>autoware_cmake</buildtool_depend>
 
   <depend>autoware_adapi_v1_msgs</depend>
+  <depend>tier4_control_msgs</depend>
 
   <test_depend>ament_lint_auto</test_depend>
-  <test_depend>autoware_lint_common</test_depend>
```
Comment (isamu-takagi) [inline common/component_interface_specs/package.xml:19]: This dependency is required for linting header files.

File: common/autoware_ad_api_specs/package.xml [hunk 2]
```diff
@@ -13,6 +13,8 @@
   <buildtool_depend>ament_cmake_auto</buildtool_depend>
   <buildtool_depend>autoware_cmake</buildtool_depend>
 
+  <depend>autoware_adapi_version_msgs</depend>
```
Comment (isamu-takagi) [inline common/autoware_ad_api_specs/package.xml:16]: No dependencies should be written on this package. This is because packages that use this package will have unnecessary dependencies. The packages that use this package write dependencies according to the headers that they use.

Comment (isamu-takagi) [reply common/autoware_ad_api_specs/package.xml:16]: Sorry. This comment is for https://github.com/autowarefoundation/autoware.universe/pull/3606#discussion_r1199956126.

File: common/component_interface_specs/package.xml (hunk 1 above)
Comment (VRichardJP) [reply common/component_interface_specs/package.xml:19]: Looks like a bad copy/paste from my side
Thanks!

**Review by isamu-takagi — APPROVED** [review]

**Comment by mitsudome-r** [comment]
Since this PR is too big, maybe it is better to split into smaller PRs (e.g., by top-level folder: perception, localization, sensing, etc.)(

**Comment by YamatoAndo** [comment]
@VRichardJP @mitsudome-r For the localization part, I have created a [new PR](https://github.com/autowarefoundation/autoware.universe/pull/5171)

**Comment by YamatoAndo** [comment]
@VRichardJP @mitsudome-r Also about map packages https://github.com/autowarefoundation/autoware.universe/pull/5172

**Comment by kminoda** [comment]
@VRichardJP What is the status of this PR?

**Comment by VRichardJP** [comment]
@kminoda 
As @mitsudome-r has said, this PR is too big. I thought it was simpler to update all packages.xml at once because it would not be necessary to deal with downstream dependency issues. Since it just impact compilation "if it still compiles it should work™". But this requires too many reviewers...

As @YamatoAndo has started to do, working at module level is the way to do. Unfortunatelly, I am busy with other projects with days and I cannot find the time to contribute much.

**Comment by kminoda** [comment]
@VRichardJP Understood, thank you! Would you mind making this PR as a draft?
